### ISSUE TITLE
pyright fails to respect hasattr

### ISSUE THREAD

**Issue by boompig** [microsoft/pyright#10890]
Sample python code

```python
class Subscription:
    pass

class Plan:
    pass


subscription = Subscription()
assert hasattr(subscription, "plan") and isinstance(subscription.plan, Plan)

print(subscription.plan)
```

This will error under pyright latest (pyright 1.1.404) with python3.12 or python3.13

Error:

```
/tmp/foo.py
  /tmp/foo.py:9:66 - error: Cannot access attribute "plan" for class "Subscription"
    Attribute "plan" is unknown (reportAttributeAccessIssue)
  /tmp/foo.py:11:20 - error: Cannot access attribute "plan" for class "Subscription"
    Attribute "plan" is unknown (reportAttributeAccessIssue)
2 errors, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
This is by design, not a bug. The static definition of the class `Subscription` does not declare an attribute named `plan`, nor does it define a `__getattr__` method. Therefore you can't access it without triggering an error in the type checker.
